tests/integration/security/sds_vault_flow/sds_vault_flow_test.go

```diff
@@ -0,0 +1,188 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sds_vault_test
+
+import (
+	"fmt"
+	"path"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+var (
+	inst istio.Instance
+)
+
+const (
+	tlsRootCert = "-----BEGIN CERTIFICATE-----\\\\nMIIC3jCCAcagAwIBAgIRAIcSFH1jneS0XPz5r2Q" +
+		"DbigwDQYJKoZIhvcNAQELBQAw\\\\nEDEOMAwGA1UEChMFVmF1bHQwIBcNMTgxMjI2MDkwM" +
+		"DU3WhgPMjExODEyMDIwOTAw\\\\nNTdaMBAxDjAMBgNVBAoTBVZhdWx0MIIBIjANBgkqhki" +
+		"G9w0BAQEFAAOCAQ8AMIIB\\\\nCgKCAQEA2q5lfJCLAOTEjX3xV8qMLEX8zUQpd0AjD6zzO" +
+		"Mzx51GVM7Plf7CJmaDq\\\\nyloRz3zcrTEltHUrln5fvouvp4TetOlqEU979vvccnFLgXr" +
+		"Spn+Zt/EyjE0rUYY3\\\\n5e2qxy9bP2E7zJSKONIT6zRDd2zUQGH3zUem1ZG0GFY1ZL5qF" +
+		"SOIy+PvuQ4u8HCa\\\\n1CcnHmI613fVDbFbaxuF2G2MIwCZ/Fg6KBd9kgU7uCOvkbR4AtR" +
+		"e0ntwweIjOIas\\\\nFiohPQzVY4obrYZiTV43HT4lGti7ySn2c96UnRSnmHLWyBb7cafd4" +
+		"WZN/t+OmYSd\\\\nooxCVQ2Zqub6NlZ5OySYOz/0BJq6DQIDAQABozEwLzAOBgNVHQ8BAf8" +
+		"EBAMCBaAw\\\\nDAYDVR0TAQH/BAIwADAPBgNVHREECDAGhwQj6fn5MA0GCSqGSIb3DQEBC" +
+		"wUAA4IB\\\\nAQBORvUcW0wgg/Wo1aKFaZQuPPFVLjOZat0QpCJYNDhsSIO4Y0JS+Y1cEIk" +
+		"vXB3S\\\\nQ3D7IfNP0gh1fhtP/d45LQSPqpyJF5vKWAvwa/LSPKpw2+Zys4oDahcH+SEKi" +
+		"Qco\\\\nIhkkHNEgC4LEKEaGvY4A8Cw7uWWquUJB16AapSSnkeD2vTcxErfCO59yR7yEWDa" +
+		"6\\\\n8j6QNzmGNj2YXtT86+Mmedhfh65Rrh94mhAPQHBAdCNGCUwZ6zHPQ6Z1rj+x3" +
+		"Wm9\\\\ngqpveVq2olloNbnLNmM3V6F9mqSZACgADmRqf42bixeHczkTfRDKThJcpY5" +
+		"U44vy\\\\nw4Nm32yDWhD6AC68rDkXX68m\\\\n-----END CERTIFICATE-----"
+)
+
+type testPolicy struct {
+	t         *testing.T
+	env       *kube.Environment
+	namespace string
+	fileName  string
+}
+
+type connection struct {
+	from            apps.KubeApp
+	to              apps.KubeApp
+	protocol        apps.AppProtocol
+	port            int
+	expectedSuccess bool
+}
+
+func (p testPolicy) tearDown() {
+	p.t.Logf("Tearing down policy %q.", p.fileName)
+	if err := p.env.Delete(p.namespace, path.Join("testdata", p.fileName)); err != nil {
+		p.t.Fatalf("Cannot delete %q from namespace %q: %v", p.fileName, p.namespace, err)
+	}
+}
+
+func applyPolicyFile(t *testing.T, env *kube.Environment, namespace string, fileName string) *testPolicy {
+	t.Logf("Applying policy file %v", fileName)
+	if err := env.Apply(namespace, path.Join("testdata", fileName)); err != nil {
+		t.Fatalf("Cannot apply %q to namespace %q: %v", fileName, namespace, err)
+		return nil
+	}
+	return &testPolicy{
+		t:         t,
+		env:       env,
+		namespace: namespace,
+		fileName:  fileName,
+	}
+}
+
+func checkConnection(conn connection) error {
```

Comment:
Is this method a duplicate of the one above? Any chance that you can re-use somehow?

Reply:
Done: changed to re-use this method.